CoreNotesView/src/au/gov/asd/tac/constellation/views/notes/utilities/MarkdownNode.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2010-2022 Australian Signals Directorate
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.views.notes.utilities;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is the MarkdownNode and represents the different types of text
+ * that is processed
+ *
+ * @author altair1673
+ */
+public class MarkdownNode {
+
+    // Can have many children
+    private final List<MarkdownNode> children = new ArrayList<>();
+
+    private int startIndex = -99;
+    private int endIndex = -99;
+
+    private String value = "";
+
+    private int tabs = 0;
+    private int latestListItem = 1;
+
+    private Type type;
+
+    private int headingLevel = -99;
+
+    public enum Type {
+        ROOT,
+        HEADING,
+        PARAGRAPH,
+        BOLD,
+        ITALIC,
+        LINE_BREAK,
+        STRIKETHROUGH,
+        ORDERED_LIST,
+        LIST_END,
+        LIST_ITEM,
+        NORMAL
+    }
+
+    public MarkdownNode() {
```

Comment:
I would set your default values for all your fields here in the default constructor rather than above